clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -723,12 +732,20 @@ protected synchronized void markCoordinatorUnknown(boolean isDisconnected) {
      * Get the current generation state if the group is stable.
      * @return the current generation or null if the group is unjoined/rebalancing
      */
-    protected synchronized Generation generation() {
+    protected synchronized Generation generationOnGroupStable() {
```

Comment:
`AbstractCoordinator` is an unofficially exposed API. We generally don't mind breaking it, but we try to avoid it if we can. I wonder if we could leave `generation()` as it was and introduce alternative getters with a different name? Maybe if we just had a `memberId()` and `hasValidGenerationId()`, that would be good enough for testing?